### ISSUE TITLE
Optimise error message for workspace name

### ISSUE THREAD

**Issue by chrifro** [coder/coder#14824]
**Issue**
It's tricky to decipher the error message and to understand what I need to change in order to fix it
![image](https://github.com/user-attachments/assets/a42f237d-deb6-4901-ae30-7c8b41399c83)

**Possible solution:**
 Update message to “Special characters like #+*”$ are not supported”

**Comment by mtojek** [comment]
@chrifro I agree the error message is awkward, should we mention ALL allowed special characters or rather switch to "Special characters are not supported" without mentioning every single one?

**Comment by chrifro** [comment]
Let's go with a mix:  "Special characters (e.g., !, @, #) are not supported"
